app/routes/app/rewards/index.tsx

```diff
@@ -27,25 +27,29 @@ import { ClaimRewardWeb3Button } from "~/features/web3-button/claim-reward";
 import invariant from "tiny-invariant";
 import type { SendTransactionResult } from "@wagmi/core";
 import { defaultNotifyTransactionActions } from "~/features/web3-transaction-toasts";
+import { searchSubmissions } from "~/services/submissions.server";
+import type { SubmissionDoc } from "~/domain/submission";
 
 export const loader = async (data: DataFunctionArgs) => {
-  const user = await getUserId(data.request);
-  const wallets = user ? await findAllWalletsForUser(user) : [];
+  const user = await getUser(data.request);
+  invariant(user, "Could not find user, please sign in");
+  const wallets = await findAllWalletsForUser(user.id);
+  const submissions = await searchSubmissions({
```

Comment:
I think you can get away with only passing in serviceProvider and the rest will default to a value right?

Reply:
That was givig me the red squiggle

Reply:
Also could just add filter sort to this branch since doing that would mean these get passed in anyway

Reply:
do it!